cow_instrument/FlatTree.h

```diff
@@ -0,0 +1,75 @@
+#ifndef FLATTREE_H
+#define FLATTREE_H
+
+#include <memory>
+#include <vector>
+#include <map>
+#include <Eigen/Core>
+#include "IdType.h"
+#include "ComponentInfo.h"
+
+class Detector;
+class PathComponent;
+class PointSource;
+class PointSample;
+
+/*
+ Flattened representation of an Instrument Tree.
+ */
+class FlatTree {
```

Comment:
Indeed. That is correct. I gradually grew `ComponentInfo` and shrunk `FlatTree` (formally `InstrumentTree`). I think one or the other could disappear now. The thing that I originally liked about `ComponentInfo` is that it could be responsible for taking a real tree and populating all the necessary flattened arrays we needed.